examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+        (e.g. "requirements.txt")
+    extra_pip_requirements : Union[Iterable, str], optional (default=None)
+        Either an iterable of pip requirement strings
+        (e.g. ["pandas", "-r requirements.txt", "-c constraints.txt"]) or the string
+        path to a pip requirements file on the local filesystem
+        (e.g. "requirements.txt")
+    serialization_format : str, optional (default="pickle")
+        The format in which to serialize the model. This should be one of the formats
+        "pickle" or "cloudpickle"
+    kwargs:
+        Additional arguments for :py:class:`mlflow.models.model.Model`
+
+    Returns
+    -------
+    A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+    metadata of the logged model.
+    """
+    return Model.log(
+        artifact_path=artifact_path,
+        flavor=flavor,
+        registered_model_name=registered_model_name,
+        sktime_model=sktime_model,
+        conda_env=conda_env,
+        code_paths=code_paths,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        serialization_format=serialization_format,
+        **kwargs,
+    )
+
+
+def load_model(model_uri, dst_path=None):
+    """
+    Load a sktime model from a local file or a run.
+
+    Parameters
+    ----------
+    model_uri : str
+        The location, in URI format, of the MLflow model. For example:
+
+                    - ``/Users/me/path/to/local/model``
+                    - ``relative/path/to/local/model``
+                    - ``s3://my_bucket/path/to/model``
+                    - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                    - ``mlflow-artifacts:/path/to/model``
+
+        For more information about supported URI schemes, see
+        `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+        artifact-locations>`_.
+    dst_path : str, optional (default=None)
+        The local filesystem path to which to download the model artifact.This
+        directory must already exist. If unspecified, a local output path will
+        be created.
+
+    Returns
+    -------
+    A sktime model instance.
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+    _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
+    sktime_model_file_path = os.path.join(local_model_path, flavor_conf["pickled_model"])
+    serialization_format = flavor_conf.get("serialization_format", SERIALIZATION_FORMAT_PICKLE)
+    return _load_model(path=sktime_model_file_path, serialization_format=serialization_format)
+
+
+def _save_model(model, path, serialization_format):
```

Comment:
Is there any particular reason for using pickle here? A cursory look at the wrappers in sktime seems to indicated that they do expose the native serialization / deserialization protocols for the underlying models (here's Prophet's: https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.forecasting.fbprophet.Prophet.html#sktime.forecasting.fbprophet.Prophet.save )

Reply:
sktime uses pickle as the default serialization (https://github.com/sktime/sktime/blob/main/sktime/base/_base.py#L618-L664), this was not questioned or topic of discsssion as part of the mlflow custom flavor PR

Reply:
fair enough. I'd just add a note to the example that explains the limitations (you must use the same python environment (version) in whatever environment you're going to use this for inference to ensure that the model will load with appropriate version of pickle)

Reply:
added a comment in the code as you suggested